src/hooks/useIntersectionObserver.ts

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useRef, useState } from 'react'
+
+const useIntersectionObserver = () => {
+  const observerRef = useRef<HTMLDivElement>(null)
+  const [observerIsSet, setObserverIsSet] = useState(false)
+  const [isIntersecting, setIsIntersecting] = useState(false)
+
+  useEffect(() => {
+    const checkObserverIsIntersecting = ([entry]: IntersectionObserverEntry[]) => {
+      setIsIntersecting(entry.isIntersecting)
+    }
+
+    if (!observerIsSet) {
+      const intersectionObserver = new IntersectionObserver(checkObserverIsIntersecting, {
+        rootMargin: '0px',
+        threshold: 1,
+      })
+      intersectionObserver.observe(observerRef.current)
+      setObserverIsSet(true)
+    }
+  }, [observerIsSet])
+
+  return { observerRef, isIntersecting }
+}
+
+export default useIntersectionObserver
```

Comment:
`ref` never gets set. I think you need to pass it in.

Reply:
It is not set in here, we pass ref object to the callers of that hook and they should add this to a DOM node, For example;

https://github.com/pancakeswap/pancake-frontend/blob/0a2c6d7a74a9842514451933ae7298389908f760/src/views/Home/components/EarnAPRCard.tsx#L89